src/fv/share/tdyugrid.c

```diff
@@ -0,0 +1,1209 @@
+#include <petscdm.h>
+#include <petscdmda.h>
+#include <petscviewerhdf5.h>
+#include <private/tdyugridimpl.h>
+#include <private/tdymemoryimpl.h>
+#include <private/tdyutils.h>
+
+static PetscInt vertex_separator = -777;
+static PetscInt dual_separator = -888;
+static PetscInt cell_separator = -999999;
+
+/// Allocates memory and initializes a TDyUGrid struct
+///
+/// @param [out] ugrid A TDyUGrid struct
+///
+/// @returns 0 on success, or a non-zero error code on failure
+PetscErrorCode TDyUGridCreate(TDyUGrid **ugrid) {
+
+  PetscErrorCode ierr;
+
+  ierr = PetscCalloc(sizeof(TDyUGrid), ugrid); CHKERRQ(ierr);
+
+  (*ugrid)->num_cells_global = 0;
+  (*ugrid)->num_cells_local = 0;
+  (*ugrid)->num_cells_ghost = 0;
+
+  (*ugrid)->max_verts_per_cell = 0;
+  (*ugrid)->max_ndual_per_cell = 0;
+
+  (*ugrid)->num_verts_global = 0;
+  (*ugrid)->num_verts_local = 0;
+
+  (*ugrid)->global_offset = 0;
+
+  (*ugrid)->cell_vertices = NULL;
+  (*ugrid)->cell_num_vertices = NULL;
+
+  (*ugrid)->cell_ids_natural = NULL;
+  (*ugrid)->cell_ids_petsc = NULL;
+  (*ugrid)->ghost_cell_ids_petsc = NULL;
+
+  (*ugrid)->cell_neighbors_local_ghosted = NULL;
+  (*ugrid)->cell_num_neighbors_local_ghosted = NULL;
+
+  (*ugrid)->vertices = NULL;
+
+  (*ugrid)->ao_natural_to_petsc = NULL;
+
+  (*ugrid)->max_cells_sharing_a_vertex = 24;
+  (*ugrid)->max_vert_per_face = 4;
+  (*ugrid)->max_face_per_cell = 6;
+
+  (*ugrid)->face_to_vertex_natural = NULL;
+  (*ugrid)->face_to_vertex = NULL;
+  (*ugrid)->cell_to_face_ghosted = NULL;
+  (*ugrid)->vertex_ids_natural = NULL;
+
+  PetscFunctionReturn(0);
+}
+
+/// Read in a PFLOTRAN-formatted HDF5 mesh file
+///
+/// @param [in] mesh_file Name of the mesh file
+/// @param [inout] ugrid A TDyUGrid struct
+///
+/// @returns 0 on success, or a non-zero error code on failure
+static PetscErrorCode ReadPFLOTRANMeshFile(const char *mesh_file, TDyUGrid *ugrid){
+
+  PetscViewer viewer;
+  Vec cells, verts;
+  PetscScalar *v_p;
+  PetscErrorCode ierr;
+
+  PetscInt ***cell_vertices = &ugrid->cell_vertices;
+  PetscReal ***vertices = &ugrid->vertices;
+  PetscInt *num_cells_local = &ugrid->num_cells_local;
+  PetscInt *max_verts_per_cell = &ugrid->max_verts_per_cell;
+  PetscInt *num_verts_local = &ugrid->num_verts_local;
+
+  ierr = PetscViewerHDF5Open(PETSC_COMM_WORLD, mesh_file, FILE_MODE_READ, &viewer);CHKERRQ(ierr);
+
+  ierr = PetscViewerHDF5PushGroup(viewer, "/Domain");CHKERRQ(ierr);
+
+  // Read cells
+  ierr = VecCreate(PETSC_COMM_WORLD, &cells); CHKERRQ(ierr);
+  ierr = PetscObjectSetName((PetscObject) cells, "cells");CHKERRQ(ierr);
+  ierr = PetscObjectSetName((PetscObject) cells, "Cells"); CHKERRQ(ierr);
```

Comment:
I don't think you can give two names to any single `PetscObject`, so line 87 is likely overwriting the effects of line 86 here. I'd just pick one name and go with it.